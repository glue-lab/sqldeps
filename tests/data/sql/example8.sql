-- Postgres Function
CREATE OR REPLACE FUNCTION web_import."build_Api_Property_Defor"()
 RETURNS void
 LANGUAGE plpgsql
AS $function$BEGIN
	TRUNCATE TABLE web_import."Api_Property_Defor";

INSERT INTO web_import."Api_Property_Defor"(
		"PropertyId", "Year", "Ha"
	)
	SELECT ps."PropertyId", d."Year", avg("Defor") AS "Ha"
FROM build_public."Property_Shape" ps
INNER JOIN (
				SELECT "ShapeId", "Year"::INTEGER, SUM("areaha") AS "Defor"
FROM build_spatial."Shape_Defor"
WHERE "Year"::text ~ '^[0-9]+$'
-- and "areaha">6.25
GROUP BY "ShapeId", "Year"::INTEGER
			) d
				ON
d."ShapeId" = ps."ShapeId"
WHERE ps."PropertyId" IS NOT NULL
GROUP BY ps."PropertyId", d."Year";

END
$function$