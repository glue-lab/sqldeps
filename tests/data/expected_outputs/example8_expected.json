{
    "dependencies": {
        "build_public.Property_Shape": ["PropertyId", "ShapeId"],
        "build_spatial.Shape_Defor": ["ShapeId", "Year", "areaha"],
        "web_import.Api_Property_Defor": ["Ha", "PropertyId", "Year"]
    },
    "outputs": {
      "web_import.Api_Property_Defor": ["Ha", "PropertyId", "Year"]
    }
}